### PR TITLE
First commit of Fortran version of GildedRose

### PR DESCRIPTION
**Pull request by pclausen**
First commit of Fortran version of GildedRose

I followed the C version the best I could but I could not make the C-version build so I compared the output to the python version and it looked OK.

I hope to use this code kata some day at my company where we still happily use Fortran for numerical stuff aka HPC. Fortran is not the coolest language but still used and for numerics it is still hard to beat. :-)

### CODE REVIEW

File: fortran/test/GildedRose_unity_test.f90 [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+
+
+
+program GildedRose_texttest
+  use GildedRose
+  implicit none
+  type(Item) :: it
+
+  call init_item(it, "foo", 10, 20)
```
Comment (emilybache) [inline fortran/test/GildedRose_unity_test.f90:12]: I wonder if there should be a call here to update_quality before the assertion?

**Comment by emilybache** [comment]
Thankyou so much for this! Could you just check the unit test - I think it should call update_quality somewhere?

File: fortran/test/GildedRose_unity_test.f90 (hunk 1 above)
Comment (pclausen) [reply fortran/test/GildedRose_unity_test.f90:12]: You are right, I missed that. Not quite sure what this test is doing but that is kind of the point, right. ;-)
Thanks for maintaining this valuable resource.

**Comment by pclausen** [comment]
Adding missing call to update_quality in GildedRose_unity_test

**Comment by emilybache** [comment]
Thankyou! This looks great.

**Pull request merged**